plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnWithAliasFormatterLegacy.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+public class ColumnWithAliasFormatterLegacy
+        implements ColumnWithAliasFormatter
+{
+    @Override
+    public JdbcColumnHandle format(JdbcColumnHandle column, int nextSyntheticColumnId, ConnectorSession session)
```

Comment:
Yes, you are right. I did check the number of occurrences:

<img width="746" alt="Screenshot 2023-08-29 at 14 14 10" src="https://github.com/trinodb/trino/assets/13519572/c3732ba5-5b2e-438d-8fc3-0068b15ea6e0">
<img width="1226" alt="Screenshot 2023-08-29 at 14 14 48" src="https://github.com/trinodb/trino/assets/13519572/da04791a-6b88-4613-b6dc-529f8582636c">

Which shows that ConnectorSession is only used as a NON-first parameter in less than 5% of cases.

Can you let me know how does it improve readability or in general management of the source code in trino if I align with 95% of cases? My concern is that the remaining 5% of the cases will not disappear, so the 0.3% of cases my change is throwing in (or not, if I make a change like you requested), seems like a marginal value.

Having said that, I can make that change, but the point of this exercise is unclear to me. It would be clearer if you requested that I change all the other occurrences by moving the ConnectorSession to the first param and possibly write a CI/CD plugin that would enforce that.

Can you justify/back up your request with numbers?

Reply:
We default to the majority case. There is no science/numbers/rules behind it - just a preference to keep it consistent if possible.

Reply:
keeping consistency has benefits. e.g. avoids easy mistakes when refactoring to use delegate and argument order is switched.

It's not 100% consistent only because there's no easy tools to do so.

Reply:
@hashhar, @wendigo - thank you for responding. You might find my questions awkward, that is because I'm trying to learn by example, and given the fact that I asked for a specific rule/document/number that would back up your request, and I did not receive it, that learning will be harder than I expected.

The reason why I'm digging on this one (even though this discussion seems pointless on the first sight), is it looks to me like I have a sample of TWO examples and it seems like one contradicts the other. I know this is 'statistically unstable' as the sample is too small, but still I need to somehow learn/make progress.

So the other counter-example is in this PR: https://github.com/trinodb/trino/pull/18718 where both of you commented and it seems like it's the other way round. That is, a change was proposed to standardise and use a builder instead of a constructor call, but that comment:

"LGTM % how do we prevent new usages of the constructor from appearing?"

by @hashhar, is what I believe stopped the PR from being merged, even though it had a green build and an approval, what I believed were the only things necessary for a merge to master to occur.

So with that long comment I wanted to say that I do not understand why:
- in this PR you are asking to 'standardise' even though there is no change that this will ever be standardised
- in the other PR I did try to standardise, and you left me in vain with the PR claiming that it could never be standardised

I appreciate your patience on reading this.

Reply:
in this PR it's just about "code-style".

the other PR will affect correctness since if a caller doesn't use the builder then the length limit value won't be set hence there's correctness which comes before style and consistency.

Reply:
I'm afraid I'm going to have to be the one that parks this discussion and does the change despite the fact that it does not look like there is a right-wrong kind of approach you both could defend.

I think we came pretty far from what we wanted to achieve and the result of carrying on is that:
a) @wendigo has even lost interest in commenting here
b) I seem to be making a conclusion that there is a rule that says:
   1. the maintainer is always right
   2. even if the maintainer is not right, look at pt 1.
which is probably exactly the opposite of why the review is being done in the first place.
   
Let me do that change and use your energy in the other place where I have a bigger problem, where I don't know how to pass a config value to a test.

Reply:
> It would be clearer if you requested that I change all the other occurrences by moving the ConnectorSession to the first param and possibly write a CI/CD plugin that would enforce that.

This is what we'd do in a perfect world. I don't think however that it can be done  in a reasonable amount of time (i.e. Less time than it takes to send a manual PR for aligning argument order) from past experiences we have writing errorprone or modernizer checks.